_posts/en/newsletters/2025-12-19-newsletter.md

```diff
@@ -0,0 +1,1292 @@
+  enables reactive security vaults, and a wide array of other contracts by
+  constraining the `scriptPubKey` and amount of an input or output in certain
+  ways. The `OP_VAULT` proposal was [withdrawn][vault withdrawn] in favor of
+  CCV as well. For more on CCV's applications, see [MATT][topic MATT].
+
+  A set of 64-bit arithmetic opcodes were [proposed][64bit bip]. Bitcoin's
+  current math operations are (surprisingly) not able to operate on the full
+  range of Bitcoin input and output amounts. Combined with other opcodes to
+  access and/or constrain input/output amounts, these expanded arithmetic
+  operations could enable new Bitcoin wallet functionality.
+
+  A [variant][txhash sponsors] of [`OP_TXHASH`][txhash] would enable
+  [transaction sponsorship][topic fee sponsorship].
+
+  Developers proposed two options for giving Script elliptic curve
+  cryptographic operations other than `OP_CHECKSIG` and related operations. One
+  [proposes][tweakadd] `OP_TWEAKADD` to enable constructing taproot
+  `scriptPubKeys`. The other [proposes][ecmath] more granular elliptic curve
+  opcodes such as `EC_POINT_ADD` motivated by similar functionality, but with
+  more potential applications such as new signature verifications or multi-signature
+  functionality. Either of these proposals could be combined with `OP_TXHASH`
+  and 64-bit arithmetic (or similar opcodes) to enable functionality similar
+  to CCV.
+
+{:#scriptrestoration}
+- **Script Restoration:** A series of 4 BIPs were [posted][gsr bips] for the
+  Script Restoration project. The Script changes and opcodes proposed in these
+  4 BIPs would enable all of the functionality proposed in the above opcode
+  proposals while allowing even more script expressivity.
+
+</div>
+
+## July
+
+{:#ccdelegation}
+- **Chain code delegation:** Jurvis Tan [posted][jt delegation] about his work
+  with Jesse Posner on a method (now called [Chain Code Delegation][BIPs
+  #2004]/BIP89) where the customer, rather than the partially trusted
+  collaborative custody provider, generates (and keeps private) the [BIP32][]
+  chain code to derive child keys from the provider's signing key. This way,
+  the provider cannot derive the customer's full key tree. The method can be
+  used either blinded (for complete privacy while still leveraging the
+  provider's key security) or non-blinded (allowing the provider to enforce
+  policy at the cost of revealing the specific transactions being signed to
+  the provider).
+
+## August
+
+{:#utreexo}
+- **Utreexo draft BIPs:** Calvin Kim, Tadge Dryja and Davidson Souza co-authored
+  [three draft BIPs][news366 utreexo] about a new alternative to the UTXO set, called
+  [Utreexo][topic utreexo], for nodes to obtain and verify information about UTXOs
+  being spent in a transaction. The proposal would make use of a merkle tree to
+  accumulate references to every UTXO allowing nodes to avoid storing the outputs.
```

Comment:
```suggestion
  being spent in a transaction. The proposal would make use of a forest of merkle trees to
  accumulate references to every UTXO allowing nodes to avoid storing the outputs.
```